src/kernel/mem/kheap.c

```diff
@@ -11,15 +11,27 @@ KHeapMemoryNode *kernel_heap_end;
 extern unsigned int end_of_mapped_memory;
 
 void initialize_kheap(){
-    //That should be mapped? 
+    #ifndef _TEST_
+    uint64_t *kheap_vaddress = vmm_alloc(PAGE_SIZE_IN_BYTES);
+    uint64_t phys_address = pmm_alloc_frame();    
+    map_phys_to_virt_addr((void*) phys_address, (void *)kheap_vaddress, PRESENT);
```

Comment:
this is strange to me: if the vmm manages the address space, it should also be doing the mapping/unmapping no?

Allocating physical memory and mapping that to the virtual memory should happen inside the vmm. The idea here is that you might not always be mapping previously unused memory like this. The vmm might be mapping a file, or shared memory for ipc. Hiding them behind the vmm API and controlling them with flags is much easier for the calling code.